File: app/src/main/java/br/com/alexandre/friday/recognize/chain/SuccessfulMessageSentChain.java
package br.com.alexandre.friday.recognize.chain;

import java.util.Arrays;
import java.util.Collection;

import br.com.alexandre.friday.recognize.RecognizerChain;
import br.com.alexandre.friday.recognize.RecognizerRequest;

public class SuccessfulMessageSentChain extends RecognizerChain {
    public Collection<String> execute(RecognizerRequest request) {
        return Arrays.asList("Mensagem enviada com sucesso!");
    }

    public boolean apply(RecognizerRequest request) {
        return (request.getQ().contains("mensagem") && request.getQ().contains("mãe"));
    }
}

File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/SuccessfulMessageSentChain.kt
package br.com.alexandre.friday.recognize.chain

import br.com.alexandre.friday.recognize.RecognizerChain
import br.com.alexandre.friday.recognize.RecognizerRequest
import java.util.*

class SuccessfulMessageSentChain: RecognizerChain(){
    override fun execute(request: RecognizerRequest?): MutableCollection<String> {
        return Arrays.asList("Mensagem enviada com sucesso!");
    }

    override fun apply(request: RecognizerRequest?): Boolean {
        return request?.q!!.contains("mensagem")
    }
}